### PR TITLE
MGMT-14564: Assisted-Installer: Fix vSphere on 4.13

### PR DESCRIPTION
**Pull request by eliorerz**
/cc @eliorerz

### CODE REVIEW

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-12-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-13

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-12-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-13

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-13

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-12-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-13

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-12-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-12-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-13

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-12-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-13

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-13

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-12-periodic

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-12-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-13

File: ci-operator/step-registry/assisted/vsphere/postinstall/assisted-vsphere-postinstall-commands.sh [hunk 1]
```diff
@@ -15,8 +15,67 @@ ssh -F ${SHARED_DIR}/ssh_config "root@ci_machine" "find \${KUBECONFIG} -type f -
...
+server = os.environ.get("VSPHERE_VCENTER")
+cluster = os.environ.get("VSPHERE_CLUSTER")
+datacenter = os.environ.get("VSPHERE_DATACENTER")
+datastore = os.environ.get("VSPHERE_DATASTORE")
+network = os.environ.get("VSPHERE_NETWORK")
+folder = os.environ.get("VSPHERE_FOLDER")
+username = os.environ.get("VSPHERE_USERNAME")
+password = os.environ.get("VSPHERE_PASSWORD")
+
+data = data.replace("vcenterplaceholder", server)
```
Comment (adriengentil) [inline ci-operator/step-registry/assisted/vsphere/postinstall/assisted-vsphere-postinstall-commands.sh:62]: why not just a sed?
```
sed -i -e "s/vcenterplaceholder/${VSPHERE_VCENTER}/g" \
-e  "s/clusterplaceholder/${VSPHERE_CLUSTER}/g" \
-e ...
infrastructures.config.openshift.io.yaml
```

File: ci-operator/step-registry/assisted/vsphere/postinstall/assisted-vsphere-postinstall-commands.sh [hunk 2]
```diff
@@ -34,26 +93,25 @@ EOF
...
-    [VirtualCenter "${VSPHERE_VCENTER}"]
-    datacenters = "${VSPHERE_DATACENTER}"
-EOF
+# Do the following if OCP version is >=4.13
+compare_versions "$version" "4.12"
+if [[ $? -eq 0 ]]; then
+    # Taint the nodes with the uninitialized taint
+    nodes=$(oc get nodes -o wide | awk '{print $1}' | tail -n +2)
+    for NODE in $nodes; do
+      oc adm taint node "$NODE" node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule
```
Comment (adriengentil) [inline ci-operator/step-registry/assisted/vsphere/postinstall/assisted-vsphere-postinstall-commands.sh:102]: shouldn't it be tainted after the config is updated? I guess the taint allows the controller to reload the configuration and get the right credentials?

Comment (eliorerz) [reply ci-operator/step-registry/assisted/vsphere/postinstall/assisted-vsphere-postinstall-commands.sh:102]: Nope, before updating the infrastructure config

Comment (adriengentil) [reply ci-operator/step-registry/assisted/vsphere/postinstall/assisted-vsphere-postinstall-commands.sh:102]: what is the reason?

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-13

**Comment by eliorerz** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-12-periodic

**Comment by eliorerz** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-vsphere-assisted-4-13

**Review by adriengentil — APPROVED** [review]

**Pull request merged**